Makefile

```diff
@@ -616,8 +616,8 @@ $(HELM):
 istio-remote.yaml: $(HELM)
 	cat install/kubernetes/templates/namespace.yaml > install/kubernetes/$@
 	$(HELM) template --namespace=istio-system \
-		  --set global.pilotIp="pilotIpReplace" \
-		  --set global.mixerIp="mixerIpReplace" \
+		  --set global.pilotEndpoint="pilotIpReplace" \
```

Comment:
Still not happy with pilotIpReplace - it assumes the istio-remote will only be used for the tests, but I think we want it for real users. 
How about using 'pilot.istio-system' as a default ?

Reply:
Costin,

I am also displeased by this solution, even though I proposed it.  The downside of using pilot.istio-system is that the remotes can't directly discover pilot.istio-system's IP address.  The general model I use is something like this:
```
# Determine pod endpoint addresses
pilotEndpoint=$(kubectl get pods -o wide -n istio-system | grep pilot | awk '{ print $6 }')
statsdEndpoint=$(kubectl get pods -o wide -n istio-system | grep statsd | awk '{ print $6 }')
policyEndpoint=$(kubectl get pods -o wide -n istio-system | grep policy | awk '{ print $6 }')

# Build an istio-remote.yaml using helm template
rm -f $HOME/istio-remote-out.yaml
helm template $ISTIO/istio/install/kubernetes/helm/istio-remote --name istio-remote --namespace istio-system --set global.pilotEndpoint=${pilotEndpoint} --set global.statsdEndpoint=${statsdEndpoint} --set global.policyEndpoint=${policyEndpoint} > $HOME/istio-remote-out.yaml
```

As you can see, this needs to be run against an existing Istio control plane....

Reply:
One way or another we'll need to customize the DNS - either by creating Service entries in the cluster, with external IPs of the pilot, or by customizing resolv.conf (https://kubernetes.io/docs/tasks/administer-cluster/dns-custom-nameservers/). 
DNS is needed for resolving all the VIPs, not only pilot.

Second, with MTLS and SNI - I think it's important to use a name that can match against the
cert ( pilot cert can be customized to have real DNS domain ). 

From the perspective of the test - it doesn't matter since you can replace istio-pilot.istio-system just as well as mixerIpReplace. 

I don't think this is a blocker for this PR - but we need to fix it eventually.

Reply:
I've adopted some of the changes you suggested here: https://github.com/istio/istio/pull/5083